lxd/agent.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+
+	"github.com/lxc/lxd/shared"
+)
+
+func vsockHTTPClient(vsockID int, tlsClientCert string, tlsClientKey string, tlsCA string, tlsServerCert string, insecureSkipVerify bool) (*http.Client, error) {
```

Comment:
Any reason we can't put this in a separate package, say `lxd/vsock`?

Reply:
Done